src/main/kotlin/org/jboss/tools/intellij/kubernetes/model/resource/kubernetes/custom/GenericResourceFactory.kt

```diff
@@ -0,0 +1,87 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.intellij.kubernetes.model.resource.kubernetes.custom
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import io.fabric8.kubernetes.api.model.ObjectMeta
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder
+import java.util.stream.Collectors
+
+object GenericResourceFactory {
+
+	private const val API_VERSION = "apiVersion"
+	private const val KIND = "kind"
+	private const val METADATA = "metadata"
+	private const val SPEC = "spec"
+	private const val CREATION_TIMESTAMP = "creationTimestamp"
+	private const val GENERATION = "generation"
+	private const val NAME = "name"
+	private const val NAMESPACE = "namespace"
+	private const val RESOURCE_VERSION = "resourceVersion"
+	private const val SELF_LINK = "selfLink"
+	private const val UID = "uid"
+
+	fun createResources(items: List<Map<String, String>>): List<GenericResource> {
+		return items.stream()
+				.map { createResource(it) }
```

Comment:
You're absolutely right, the type for items is erroneous, the correct one, which matches what one gets from the client (which is using jackson to deserialize the json into a map of <String, Object>) is 
```
fun createResources(items: List<Map<String, Any?>>): List<GenericResource>
```
My guess is that the generic parts of the map type are not used when casts are done at runtime and this thus succeeded at runtime. At compile-time this was fine for the kotlin compiler because of it is an up-cast String->Object.
Makes sense? Missing something?